perfkitbenchmarker/publisher.py

```diff
@@ -227,12 +228,16 @@ def _FormatMetadata(self, metadata):
                     for k, v in sorted(metadata.iteritems()))
 
   def PublishSamples(self, samples):
+    stream = io.BytesIO()
     dashes = '-' * 25
-    self.stream.write('\n' + dashes +
-                      'PerfKitBenchmarker Results Summary' +
-                      dashes + '\n')
+    stream.write('\n' + dashes +
```

Comment:
Clearer and less error prone for future updates + some comments :)

On Fri, Apr 24, 2015 at 3:01 PM, Connor McCoy <notifications@github.com>
wrote:

> In perfkitbenchmarker/publisher.py
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/68#discussion_r29089576>
> :
>
> >      dashes = '-' * 25
> > -    self.stream.write('\n' + dashes +
> > -                      'PerfKitBenchmarker Results Summary' +
> > -                      dashes + '\n')
> > +    stream.write('\n' + dashes +
>
> This change is in order to print the results to two locations: self.stream
> and the log. The local variable stream is used to build the full
> formatted text output, then that message (via stream.getvalue()) is
> written to self.stream and logging.debug.
>
> Would a different variable name make this clearer? Perhaps buffer?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/68/files#r29089576>
> .
>



-- 
Anthony F. Voellm (aka Tony)
Google Voice:  (650) 516-7382
<https://www.google.com/voice/b/0?pli=1#phones>
Blog: http://perfguy.blogspot.com/